src/modules/watch/service.js

```diff
@@ -89,11 +106,18 @@ module.exports = class WatchService {
       data
     })
     await this.updateCheckedAt(watch._id)
-    return watch // this should be aggregatedWatch, need to be fixed later
+    const aggregatedWatch = await this.aggregate(watch)
+    return aggregatedWatch
   }
 
   async getByUserID(id) {
     const userID = new ObjectID(id)
-    return this.watchCollection.find({ userID }).toArray() // this should be aggregatedWatches, need to be fixed later
+    const listWatch = await this.watchCollection.find({ userID }).toArray()
+    const listAggregatedWatch = []
+    for (const watch of listWatch) {
+      const aggregatedWatch = await this.aggregate(watch)
+      listAggregatedWatch.push(aggregatedWatch)
+    }
+    return listAggregatedWatch
```

Comment:
Vẫn là vấn đề ông để await trong vòng lặp này.
Nên viết như dưới này nhé, nó sẽ cải thiện hiệu năng tương đối. Nếu chưa hiểu tại sao thì google chút về mấy thứ asynchronous trong vòng lặp nhé.
```suggestion
    const watches = await this.watchCollection.find({ userID }).toArray()
    return Promise.all(watches.map((watch) => this.aggregate(watch)))
```